### ISSUE TITLE
Ecoregions 1- 4 are NOT symbolized at all or correctly (use EPAs legends to get colors and labels).

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/RiverscapesXML#558]

**Comment by wally-mac** [comment]
@leallysmith  I'm assigning this to you. Please delegate as needed. Thanks.

**Comment by Cashe93** [comment]
@joewheaton, The issue with the Ecoregion 3 layer is that the attribute values from the RSContext layer are inconsistent with the EPA layer (luckily, this only appears to be an issue with L3). The attribute the symbology is referencing for L3 should start with a three-decimal code (e.g. 2.9.3), but for whatever reason, the level 3 ecoregion attribute that is within RSContext projects is just a two-digit integer. 

![image](https://user-images.githubusercontent.com/49880208/231904969-72a63811-d979-47ac-bc85-476d3e748f0c.png)

The only solution I can think of is to download the full North American layer and manually curate the symbology using an attribute that only references the name and not the code. This wouldn't take too much time if you were ok with an approximate similarity to the EPA symbology. If you wanted the symbology to match the EPAs exactly, that will take quite a bit longer given the number of classes.  Thoughts? I'm also open to any ideas other @Riverscapes/riverscape-project-curators might have to get the current symbology to work.

**Comment by joewheaton** [comment]
Thank you so much @Cashe93 for the explanation. @philipbaileynar what do you think. Add these to the National dataset?

**Comment by jtgilbert** [comment]
This is already pulled from the National dataset. I think it's just a data consistency issue.

**Comment by joewheaton** [comment]
I'm not following on your suggestion? We need to mitigate against know inconsistency if we can.

**Comment by shelbysawyer** [comment]
I was working in [this RS Context project](https://warehouse.riverscapes.net/p/f0a9341c-6211-43d2-a660-4d7bab3b9268/) in QRV and noticed that Level 1, 2, and 4 Ecoregions show up and are symbolized, but Level 3 Ecoregions do not. It also looks like the labels are displaying twice for Level 1 Ecoregions:

![rsc](https://github.com/Riverscapes/RiverscapesXML/assets/73311751/15ab595d-9793-4257-9c6c-dbb6561db1c6)

**Comment by Cashe93** [comment]
Just circling back on this. @shelbysawyer thanks for the notes; the level 3 ecoregion issue has been discussed above. It looks like the redundant labels are just a result of the layer being a multipart feature; each feature is labeled once but there are multiple features with the same name. 

@joewheaton and @jtgilbert how would you like me to proceed with the level 3 symbology? Also, level 4 ecoregions have nearly 1000 categories; symbology has been applied for Q and Arc but to manually do this in Mapbox is going to be pretty tedious and time-consuming. Do you think this is worth muscling through, or should we explore alternative options?

**Comment by MattReimer** [comment]
As for the duplicated label you're right that it's probably due to the multipart feature. There's no hard solution to this that I'm aware of but you might be able to dial that in (in the web anyway) using features like symbol spacing:

- [symbol-spacing](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-symbol-spacing)
- [text-allow-overlap](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#layout-symbol-text-allow-overlap)

As for the 1000 category feature:

If you symbolize one category the way you like it then it might be possible to write a python script to convert from the qml. We do something similar for the [EVT rasters HERE](https://github.com/Riverscapes/RiverscapesXML/blob/master/python/jupyter/veg_rasters/evt_bps.ipynb) but the idea is the same. You just need to output a JSON file instead of a CSV file.

**Comment by CHafen** [comment]
Just bring this back up @Cashe93 @jtgilbert @joewheaton  as level 3 ecoregions are not symbolized yet for web.

> @joewheaton and @jtgilbert how would you like me to proceed with the level 3 symbology? Also, level 4 ecoregions have nearly 1000 categories; symbology has been applied for Q and Arc but to manually do this in Mapbox is going to be pretty tedious and time-consuming. Do you think this is worth muscling through, or should we explore alternative options? @MattReimer, any ideas on how to streamline this?

**Comment by joewheaton** [comment]
I can't remember what the problem was? @Cashe93 do you? I think we figured out why it was not displaying properly, but maybe not a solution?

**Comment by joewheaton** [comment]
Also @CHafen did you see the new webRV stuff that @MattReimer added for vector layer editing in browser? It is really nice for testing and customizing what you want.

**Comment by Cashe93** [comment]
@joewheaton, yeah we figured out why it wasn't displaying properly, but still haven't pinned down a solution. See [here](https://github.com/Riverscapes/RiverscapesXML/issues/558#issuecomment-1507739138) (or above) for the comment laying out the issue.

**Comment by joewheaton** [comment]
@CHafen, can you please schedule a time to meet with @Cashe93 and maybe he can walk you through the manual fix proposed. If necessary, bring in @Jdgilby if any additional cleaning of that dataset is needed.
